scripts/build_components.sh

```diff
@@ -4,39 +4,61 @@ set -e
 function usage {
   echo "Usage: $0 [options]"
   echo "Options:"
-  echo "  -t, --tag=<value>        Tag to add to image, repeatable
-                                   The first tag is set in the component specifications"
-  echo "  -h, --help               Display this help message"
+  echo "  -t,  --tag <value>                 Tag to add to image, repeatable
+                                             The first tag is set in the component specifications"
+  echo "  -c,  --cache <value>               Use registry caching when building the components (default:false)"
+  echo "  -d,  --component-dirs <value>      Directory containing components to build as subdirectories.
+                                             The path should be relative to the root directory (default:components)"
+  echo "  -n, --namespace <value>            The namespace for the built images, should match the github organization (default: ml6team)"
+  echo "  -co, --component <value>           Specific component to build. Pass the component subdirectory name(s) to build
+                                             certain component(s) or 'all' to build all components in the components
+                                             directory (default: all)"
+  echo "  -r,  --repo <value>                Set the repo (default: fondant)"
+  echo "  -h,  --help                        Display this help message"
 }
 
 # Parse the arguments
 while [[ "$#" -gt 0 ]]; do case $1 in
-  -t|--tag) tags+=("$2"); shift;;
-  -c|--cache) caching=true;;
-  -h|--help) usage; exit;;
+  -n |--namespace) namespace="$2"; shift;;
+  -d |--components-dir ) components_dir="$2"; shift;;
+  -r |--repo) repo="$2"; shift;;
+  -t |--tag) tags+=("$2"); shift;;
+  -co|--component) components+=("$2"); shift;;
+  -c |--cache) caching=true;;
+  -h |--help) usage; exit;;
   *) echo "Unknown parameter passed: $1"; exit 1;;
 esac; shift; done
 
 # Check for required argument
-if [ -z "${tags}" ]; then
+if [ ${#tags[@]} -eq 0 ]; then
```

Comment:
Why was this needed?

Reply:
both achieve the same purpose but was getting suggestion to change it in my IDE so I opted for checking if the array length is zero instead. I can revert it back if you prefer. 
![image](https://github.com/ml6team/fondant/assets/47530815/cd7b29d8-f15f-451b-95ea-18a2e94379fe)

Reply:
My question was mainly meant to learn, not to ask you to revert :sweat_smile: 
But if they're equivalent, both are fine for me.